Move to gcc 7.1

mythi opened the pull request:
Many fixes (including meta-ros GCC7.1 readiness) have been merged
so move back to follow OE-core default GCCVERSION.

pohly commented on meta-refkit-computervision/recipes-computervision/librealsense/files/0001-Add-missing-includes-for-functional.patch:
```diff
@@ -0,0 +1,42 @@
+Newer GCC versions (>=7) fail with an error if the header is not
+included an std::function is used.
+
+Upstream-status: Backport
```
Mind the spelling (Upstream-Status) and that "Backport" usually has more information about what was backported. For example:

    libxml2-fix_NULL_pointer_derefs.patch:Upstream-Status: Backport [https://git.gnome.org/browse/libxml2/commit/?id=e905f08123e4a6e7731549e6f09dadff4cab65bd]

pohly commented on meta-refkit-gateway/recipes-core/iotivity/files/0001-iotivity-adapt-to-GCC7.patch:
```diff
@@ -0,0 +1,52 @@
+as <memory>, <future>, <mutex>, and <regex>. Correct code
+should #include <functional> to define them."
+
+Upstream-status: Pending
```
Spelling again.

pohly replied:
Can we avoid the "Pending" status and submit this right away, before merging?

pohly commented on meta-refkit-core/conf/distro/include/refkit-config.inc:
```diff
@@ -47,6 +47,9 @@ REFKIT_DEFAULT_DISTRO_FEATURES += "refkit-firewall"
 REFKIT_DEFAULT_DISTRO_FEATURES += "refkit-computervision"
 
+# Reconfigure/patch upstream recipes for the gateway vision profile.
+REFKIT_DEFAULT_DISTRO_FEATURES += "refkit-gateway"
```
This might be better to have in a separate commit. It's not mentioned in the commit message either.

mythi commented:
@pohly I've addressed the review feedback. The iotivity fix was a backport after all (OTOH, it was not found in iotivity's github mirror which I looked first. after cloning the gerrit tree I noticed it).
